src/Utilities/FileSystem.hpp

```diff
@@ -0,0 +1,125 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Declares functions to do file system manipulations
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+/*!
+ * \ingroup FileSystem
+ * \brief A light-weight file system library based on POSIX.
+ *
+ * We use this library instead of a subprocess based library because OpenMPI
+ * does not support forking of processes on all systems. Since the
+ * parallelization library we use may be implemented on top of OpenMPI we
+ * take the safe route and use POSIX.
+ */
+namespace filesystem {
+
+/*!
+ * \ingroup FileSystem
+ * \brief Given a path possibly including a file name returns the path
```

Comment:
Having thought about `get_path` and `get_file_name` I'm not sure they serve much purpose, definitely not `get_path` since it's not at all clear to me what `/path/to` should return. `get_file_name` suffers from the same ambiguity there but I think makes a bit more sense. Both can be handled by simple `string.find()` type manipulations. With that, I think we should just get rid of them because they don't really offer anything that is concrete.